### DISCUSSION TITLE
M4A-GA (Floor Heating or Water/Gas Boiler) Thermostat (Tuya) Support

### DISCUSSION THREAD

**Discussion by g6094199** [arendst/Tasmota#20377] [Support request]
Labeled: M4A-GA  or M4AGA-B

**-B** respectivly **-GA**  seems to be the version. In my case  GA: Water heating

other versions:
GA: Water heating  (3A)
GB: Electric heating (16A)
GC: Gas boiler heating (3A)
GE: Water and gas boiler heating

For the white versions its a bit different. The are called M4AHGB-W, for the white electric heating version.

I ordered the version without WIFI, cause i dont have to remove the non compatible Tuya  module first. It seems very similar to the [ME81H](https://templates.blakadder.com/ME81H.html) types and was quiet cheap (~20€).

pics:
front:
![grafik](https://github.com/arendst/Tasmota/assets/37833326/bd517afc-7f87-4b77-94ec-7a8ad3ee05f6)

front powered:
![grafik](https://github.com/arendst/Tasmota/assets/37833326/95b10819-89a0-4218-bafe-6fbdcd83bd52)

back:
![grafik](https://github.com/arendst/Tasmota/assets/37833326/7655ca52-113e-42ae-8ad0-7b60f7fe2424)
![2024-01-02-15-24-44-968](https://github.com/arendst/Tasmota/assets/37833326/681d78a7-6968-48ae-9f32-0101e470ccb6)

board:
![2024-01-02-15-05-26-714](https://github.com/arendst/Tasmota/assets/37833326/25f06c04-1182-4231-9775-adee3e0de41b)

![2024-01-02-15-06-22-646](https://github.com/arendst/Tasmota/assets/37833326/d47b334f-3d11-4f42-bd11-bab4a7f44561)

[User manual](https://thermostat.guide/minco-home/minco-home-m4a-smart-home-touchscreen-programmable-thermostat-user-manual/):

[20240102153046874.pdf](https://github.com/arendst/Tasmota/files/13810883/20240102153046874.pdf)

and [here](https://thermostat.guide/earu/earu-wifi-m4ahga-b-wifi-smart-thermostat-user-manuals/)

**Comment by barbudor** [comment]
I'm confused
What's the purpose of this if it has no Wifi module ?

If you are expeting to solder-in a ESP based module, there is some risk that the Tuya MCU is flashed with a simplified firmware which just don't try to use Wifi.

**Reply by g6094199** [reply]
surprisingly in real life (and in the old days) nothing NEEDS to have wifi to work as expected ;-)

for the rest we will see. i'll add the modul and report back.

**Comment by g6094199** [comment]
ok. the tuya modul seems to be very chatty...


![Snapshot_2024-01-03_14-50-18](https://github.com/arendst/Tasmota/assets/37833326/3996c883-00aa-4ae5-a9fe-f5bd48c1ad8c)

`14:48:56.327 TYA: RX value 60 from dpId 102 
14:48:56.336 DMP: 55 AA 03 07 00 08 65 02 00 04 00 00 00 0A 86
14:48:56.339 {"TuyaReceived":{"Data":"55AA03070008650200040000000A86","Cmnd":7,"CmndData":"650200040000000A","DpType2Id101":10,"101":{"DpId":101,"DpIdType":2,"DpIdData":"0000000A"}}}
14:48:56.342 TYA: fnId=0 is set for dpId=101
14:48:56.344 TYA: RX value 10 from dpId 101 
14:48:56.349 DMP: 55 AA 03 07 00 05 67 01 00 01 00 77
14:48:56.352 {"TuyaReceived":{"Data":"55AA03070005670100010077","Cmnd":7,"CmndData":"6701000100","DpType1Id103":0,"103":{"DpId":103,"DpIdType":1,"DpIdData":"00"}}}
14:48:56.354 TYA: fnId=0 is set for dpId=103
14:48:56.361 DMP: 55 AA 03 07 00 05 6D 01 00 01 01 7E
14:48:56.364 {"TuyaReceived":{"Data":"55AA030700056D010001017E","Cmnd":7,"CmndData":"6D01000101","DpType1Id109":1,"109":{"DpId":109,"DpIdType":1,"DpIdData":"01"}}}
14:48:56.366 TYA: fnId=0 is set for dpId=109
14:48:56.375 DMP: 55 AA 03 07 00 05 6C 01 00 01 00 7C
14:48:56.377 {"TuyaReceived":{"Data":"55AA030700056C010001007C","Cmnd":7,"CmndData":"6C01000100","DpType1Id108":0,"108":{"DpId":108,"DpIdType":1,"DpIdData":"00"}}}
14:48:56.380 TYA: fnId=0 is set for dpId=108
14:48:56.386 DMP: 55 AA 03 07 00 05 6A 04 00 01 02 7F
14:48:56.388 {"TuyaReceived":{"Data":"55AA030700056A040001027F","Cmnd":7,"CmndData":"6A04000102","DpType4Id106":2,"106":{"DpId":106,"DpIdType":4,"DpIdData":"02"}}}
14:48:56.390 TYA: fnId=0 is set for dpId=106
14:48:56.398 DMP: 55 AA 03 07 00 05 6E 04 00 01 00 81
14:48:56.401 {"TuyaReceived":{"Data":"55AA030700056E0400010081","Cmnd":7,"CmndData":"6E04000100","DpType4Id110":0,"110":{"DpId":110,"DpIdType":4,"DpIdData":"00"}}}
14:48:56.403 TYA: fnId=0 is set for dpId=110
14:48:56.410 DMP: 55 AA 03 07 00 05 6B 04 00 01 01 7F
14:48:56.413 {"TuyaReceived":{"Data":"55AA030700056B040001017F","Cmnd":7,"CmndData":"6B04000101","DpType4Id107":1,"107":{"DpId":107,"DpIdType":4,"DpIdData":"01"}}}
14:48:56.415 TYA: fnId=0 is set for dpId=107
14:48:56.423 DMP: 55 AA 03 07 00 05 68 01 00 01 00 78
14:48:56.426 {"TuyaReceived":{"Data":"55AA03070005680100010078","Cmnd":7,"CmndData":"6801000100","DpType1Id104":0,"104":{"DpId":104,"DpIdType":1,"DpIdData":"00"}}}
14:48:56.428 TYA: fnId=0 is set for dpId=104
14:49:03.378 TYA: Send "55aa00000000ff"
14:49:03.396 DMP: 55 AA 03 00 00 01 01 04
14:49:03.398 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:49:03.400 TYA: Heartbeat
14:49:09.773 WIF: Checking connection...
14:49:10.794 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 84 9E
14:49:10.797 {"TuyaReceived":{"Data":"55AA0307000803020004000000849E","Cmnd":7,"CmndData":"0302000400000084","DpType2Id3":132,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000084"}}}
14:49:10.800 TYA: fnId=0 is set for dpId=3
14:49:10.802 TYA: RX value 132 from dpId 3 
14:49:14.417 TYA: Send "55aa00000000ff"
14:49:14.427 DMP: 55 AA 03 00 00 01 01 04
14:49:14.430 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:49:14.431 TYA: Heartbeat
14:49:25.376 TYA: Send "55aa00000000ff"
14:49:25.388 DMP: 55 AA 03 00 00 01 01 04
14:49:25.390 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:49:25.392 TYA: Heartbeat
14:49:25.706 DMP: 55 AA 03 1C 00 00 1E
14:49:25.708 {"TuyaReceived":{"Data":"55AA031C00001E","Cmnd":28}}
14:49:25.726 TYA: Send "55aa001c0008011801030e3119039b"
14:49:25.728 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 84 9E
14:49:25.730 {"TuyaReceived":{"Data":"55AA0307000803020004000000849E","Cmnd":7,"CmndData":"0302000400000084","DpType2Id3":132,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000084"}}}
14:49:25.732 TYA: fnId=0 is set for dpId=3
14:49:25.733 TYA: RX value 132 from dpId 3 
14:49:29.773 WIF: Checking connection...
14:49:34.754 HTP: Main Menu
14:49:36.376 TYA: Send "55aa00000000ff"
14:49:36.392 DMP: 55 AA 03 00 00 01 01 04
14:49:36.394 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:49:36.395 TYA: Heartbeat
14:49:37.449 HTP: Console
14:49:40.647 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 84 9E
14:49:40.650 {"TuyaReceived":{"Data":"55AA0307000803020004000000849E","Cmnd":7,"CmndData":"0302000400000084","DpType2Id3":132,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000084"}}}
14:49:40.653 TYA: fnId=0 is set for dpId=3
14:49:40.655 TYA: RX value 132 from dpId 3 
14:49:46.392 TYA: Send "55aa001c0008011801030e312e03b0"
14:49:47.391 TYA: Send "55aa00000000ff"
14:49:47.404 DMP: 55 AA 03 00 00 01 01 04
14:49:47.406 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:49:47.408 TYA: Heartbeat
14:49:48.408 WIF: Sending Gratuitous ARP
14:49:49.763 WIF: Checking connection...
14:49:55.569 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 84 9E
14:49:55.572 {"TuyaReceived":{"Data":"55AA0307000803020004000000849E","Cmnd":7,"CmndData":"0302000400000084","DpType2Id3":132,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000084"}}}
14:49:55.575 TYA: fnId=0 is set for dpId=3
14:49:55.577 TYA: RX value 132 from dpId 3 
14:49:58.385 TYA: Send "55aa00000000ff"
14:49:58.403 DMP: 55 AA 03 00 00 01 01 04
14:49:58.405 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:49:58.407 TYA: Heartbeat
14:50:09.374 TYA: Send "55aa00000000ff"
14:50:09.386 DMP: 55 AA 03 00 00 01 01 04
14:50:09.388 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:50:09.390 TYA: Heartbeat
14:50:09.788 WIF: Checking connection...
14:50:10.487 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 83 9D
14:50:10.490 {"TuyaReceived":{"Data":"55AA0307000803020004000000839D","Cmnd":7,"CmndData":"0302000400000083","DpType2Id3":131,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000083"}}}
14:50:10.492 TYA: fnId=0 is set for dpId=3
14:50:10.494 TYA: RX value 131 from dpId 3 
14:50:20.392 TYA: Send "55aa00000000ff"
14:50:20.407 DMP: 55 AA 03 00 00 01 01 04
14:50:20.409 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:50:20.411 TYA: Heartbeat
14:50:25.400 DMP: 55 AA 03 1C 00 00 1E
14:50:25.402 {"TuyaReceived":{"Data":"55AA031C00001E","Cmnd":28}}
14:50:25.420 TYA: Send "55aa001c0008011801030e3219039c"
14:50:25.422 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 84 9E
14:50:25.424 {"TuyaReceived":{"Data":"55AA0307000803020004000000849E","Cmnd":7,"CmndData":"0302000400000084","DpType2Id3":132,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000084"}}}
14:50:25.427 TYA: fnId=0 is set for dpId=3
14:50:25.429 TYA: RX value 132 from dpId 3 
14:50:29.764 WIF: Checking connection...
14:50:31.376 TYA: Send "55aa00000000ff"
14:50:31.388 DMP: 55 AA 03 00 00 01 01 04
14:50:31.391 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:50:31.392 TYA: Heartbeat
14:50:40.338 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 85 9F
14:50:40.341 {"TuyaReceived":{"Data":"55AA0307000803020004000000859F","Cmnd":7,"CmndData":"0302000400000085","DpType2Id3":133,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000085"}}}
14:50:40.344 TYA: fnId=0 is set for dpId=3
14:50:40.345 TYA: RX value 133 from dpId 3 
14:50:42.403 TYA: Send "55aa00000000ff"
14:50:42.419 DMP: 55 AA 03 00 00 01 01 04
14:50:42.421 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:50:42.423 TYA: Heartbeat
14:50:46.401 TYA: Send "55aa001c0008011801030e322e03b1"
14:50:48.398 WIF: Sending Gratuitous ARP
14:50:49.755 WIF: Checking connection...
14:50:53.371 TYA: Send "55aa00000000ff"
14:50:53.389 DMP: 55 AA 03 00 00 01 01 04
14:50:53.391 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:50:53.393 TYA: Heartbeat
14:50:55.258 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 85 9F
14:50:55.261 {"TuyaReceived":{"Data":"55AA0307000803020004000000859F","Cmnd":7,"CmndData":"0302000400000085","DpType2Id3":133,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000085"}}}
14:50:55.265 TYA: fnId=0 is set for dpId=3
14:50:55.266 TYA: RX value 133 from dpId 3 
14:51:04.396 TYA: Send "55aa00000000ff"
14:51:04.411 DMP: 55 AA 03 00 00 01 01 04
14:51:04.413 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:51:04.415 TYA: Heartbeat
14:51:09.780 WIF: Checking connection...
14:51:10.177 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 84 9E
14:51:10.179 {"TuyaReceived":{"Data":"55AA0307000803020004000000849E","Cmnd":7,"CmndData":"0302000400000084","DpType2Id3":132,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000084"}}}
14:51:10.182 TYA: fnId=0 is set for dpId=3
14:51:10.184 TYA: RX value 132 from dpId 3 
14:51:15.408 TYA: Send "55aa00000000ff"
14:51:15.421 DMP: 55 AA 03 00 00 01 01 04
14:51:15.423 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:51:15.425 TYA: Heartbeat
14:51:25.089 DMP: 55 AA 03 1C 00 00 1E
14:51:25.092 {"TuyaReceived":{"Data":"55AA031C00001E","Cmnd":28}}
14:51:25.109 TYA: Send "55aa001c0008011801030e3319039d"
14:51:25.111 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 84 9E
14:51:25.113 {"TuyaReceived":{"Data":"55AA0307000803020004000000849E","Cmnd":7,"CmndData":"0302000400000084","DpType2Id3":132,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000084"}}}
14:51:25.116 TYA: fnId=0 is set for dpId=3
14:51:25.118 TYA: RX value 132 from dpId 3 
14:51:26.404 TYA: Send "55aa00000000ff"
14:51:26.421 DMP: 55 AA 03 00 00 01 01 04
14:51:26.423 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:51:26.425 TYA: Heartbeat
14:51:29.785 WIF: Checking connection...
14:51:37.383 TYA: Send "55aa00000000ff"
14:51:37.402 DMP: 55 AA 03 00 00 01 01 04
14:51:37.404 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:51:37.406 TYA: Heartbeat
14:51:40.027 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 83 9D
14:51:40.030 {"TuyaReceived":{"Data":"55AA0307000803020004000000839D","Cmnd":7,"CmndData":"0302000400000083","DpType2Id3":131,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000083"}}}
14:51:40.033 TYA: fnId=0 is set for dpId=3
14:51:40.035 TYA: RX value 131 from dpId 3 
14:51:46.418 TYA: Send "55aa001c0008011801030e332e03b2"
14:51:48.378 WIF: Sending Gratuitous ARP
14:51:48.388 TYA: Send "55aa00000000ff"
14:51:48.404 DMP: 55 AA 03 00 00 01 01 04
14:51:48.406 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:51:48.408 TYA: Heartbeat
14:51:49.754 WIF: Checking connection...
14:51:54.947 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 84 9E
14:51:54.950 {"TuyaReceived":{"Data":"55AA0307000803020004000000849E","Cmnd":7,"CmndData":"0302000400000084","DpType2Id3":132,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000084"}}}
14:51:54.953 TYA: fnId=0 is set for dpId=3
14:51:54.955 TYA: RX value 132 from dpId 3 
14:51:59.378 TYA: Send "55aa00000000ff"
14:51:59.396 DMP: 55 AA 03 00 00 01 01 04
14:51:59.398 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:51:59.400 TYA: Heartbeat
14:52:09.800 WIF: Checking connection...
14:52:09.866 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 84 9E
14:52:09.868 {"TuyaReceived":{"Data":"55AA0307000803020004000000849E","Cmnd":7,"CmndData":"0302000400000084","DpType2Id3":132,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000084"}}}
14:52:09.871 TYA: fnId=0 is set for dpId=3
14:52:09.873 TYA: RX value 132 from dpId 3 
14:52:10.376 TYA: Send "55aa00000000ff"
14:52:10.386 DMP: 55 AA 03 00 00 01 01 04
14:52:10.389 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:52:10.391 TYA: Heartbeat
14:52:21.399 TYA: Send "55aa00000000ff"
14:52:21.417 DMP: 55 AA 03 00 00 01 01 04
14:52:21.419 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:52:21.422 TYA: Heartbeat
14:52:24.778 DMP: 55 AA 03 1C 00 00 1E
14:52:24.781 {"TuyaReceived":{"Data":"55AA031C00001E","Cmnd":28}}
14:52:24.798 TYA: Send "55aa001c0008011801030e3418039d"
14:52:24.800 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 84 9E
14:52:24.802 {"TuyaReceived":{"Data":"55AA0307000803020004000000849E","Cmnd":7,"CmndData":"0302000400000084","DpType2Id3":132,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000084"}}}
14:52:24.805 TYA: fnId=0 is set for dpId=3
14:52:24.807 TYA: RX value 132 from dpId 3 
14:52:29.755 WIF: Checking connection...
14:52:32.374 TYA: Send "55aa00000000ff"
14:52:32.390 DMP: 55 AA 03 00 00 01 01 04
14:52:32.392 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:52:32.394 TYA: Heartbeat
14:52:39.719 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 84 9E
14:52:39.721 {"TuyaReceived":{"Data":"55AA0307000803020004000000849E","Cmnd":7,"CmndData":"0302000400000084","DpType2Id3":132,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000084"}}}
14:52:39.724 TYA: fnId=0 is set for dpId=3
14:52:39.726 TYA: RX value 132 from dpId 3 
14:52:43.392 TYA: Send "55aa00000000ff"
14:52:43.410 DMP: 55 AA 03 00 00 01 01 04
14:52:43.412 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:52:43.414 TYA: Heartbeat
14:52:46.389 TYA: Send "55aa001c0008011801030e342e03b3"
14:52:48.391 WIF: Sending Gratuitous ARP
14:52:49.801 WIF: Checking connection...
14:52:54.413 TYA: Send "55aa00000000ff"
14:52:54.431 DMP: 55 AA 03 00 00 01 01 04
14:52:54.433 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:52:54.435 TYA: Heartbeat
14:52:54.638 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 84 9E
14:52:54.642 {"TuyaReceived":{"Data":"55AA0307000803020004000000849E","Cmnd":7,"CmndData":"0302000400000084","DpType2Id3":132,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000084"}}}
14:52:54.644 TYA: fnId=0 is set for dpId=3
14:52:54.646 TYA: RX value 132 from dpId 3 
14:52:55.395 RSL: STATE = {"Time":"2024-01-03T14:52:55","Uptime":"0T00:05:08","UptimeSec":308,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"OFF","Wifi":{"AP":1,"SSId":"PHI Guest","BSSId":"06:32:09:19:F7:67","Channel":5,"Mode":"11n","RSSI":44,"Signal":-78,"LinkCount":1,"Downtime":"0T00:00:03"}}
14:53:05.407 TYA: Send "55aa00000000ff"
14:53:05.423 DMP: 55 AA 03 00 00 01 01 04
14:53:05.425 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:53:05.427 TYA: Heartbeat
14:53:09.560 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 84 9E
14:53:09.563 {"TuyaReceived":{"Data":"55AA0307000803020004000000849E","Cmnd":7,"CmndData":"0302000400000084","DpType2Id3":132,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000084"}}}
14:53:09.566 TYA: fnId=0 is set for dpId=3
14:53:09.568 TYA: RX value 132 from dpId 3 
14:53:09.760 WIF: Checking connection...
14:53:16.395 TYA: Send "55aa00000000ff"
14:53:16.405 DMP: 55 AA 03 00 00 01 01 04
14:53:16.407 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:53:16.409 TYA: Heartbeat
14:53:24.473 DMP: 55 AA 03 1C 00 00 1E
14:53:24.476 {"TuyaReceived":{"Data":"55AA031C00001E","Cmnd":28}}
14:53:24.493 TYA: Send "55aa001c0008011801030e3518039e"
14:53:24.496 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 85 9F
14:53:24.498 {"TuyaReceived":{"Data":"55AA0307000803020004000000859F","Cmnd":7,"CmndData":"0302000400000085","DpType2Id3":133,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000085"}}}
14:53:24.501 TYA: fnId=0 is set for dpId=3
14:53:24.502 TYA: RX value 133 from dpId 3 
14:53:27.392 TYA: Send "55aa00000000ff"
14:53:27.407 DMP: 55 AA 03 00 00 01 01 04
14:53:27.409 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:53:27.412 TYA: Heartbeat
14:53:29.772 WIF: Checking connection...
14:53:38.377 TYA: Send "55aa00000000ff"
14:53:38.389 DMP: 55 AA 03 00 00 01 01 04
14:53:38.391 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:53:38.394 TYA: Heartbeat
14:53:39.412 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 85 9F
14:53:39.415 {"TuyaReceived":{"Data":"55AA0307000803020004000000859F","Cmnd":7,"CmndData":"0302000400000085","DpType2Id3":133,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000085"}}}
14:53:39.418 TYA: fnId=0 is set for dpId=3
14:53:39.420 TYA: RX value 133 from dpId 3 
14:53:46.414 TYA: Send "55aa001c0008011801030e352e03b4"
14:53:48.408 WIF: Sending Gratuitous ARP
14:53:49.418 TYA: Send "55aa00000000ff"
14:53:49.430 DMP: 55 AA 03 00 00 01 01 04
14:53:49.432 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:53:49.435 TYA: Heartbeat
14:53:49.756 WIF: Checking connection...
14:53:54.334 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 83 9D
14:53:54.338 {"TuyaReceived":{"Data":"55AA0307000803020004000000839D","Cmnd":7,"CmndData":"0302000400000083","DpType2Id3":131,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000083"}}}
14:53:54.340 TYA: fnId=0 is set for dpId=3
14:53:54.342 TYA: RX value 131 from dpId 3 
14:54:00.387 TYA: Send "55aa00000000ff"
14:54:00.401 DMP: 55 AA 03 00 00 01 01 04
14:54:00.404 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:54:00.406 TYA: Heartbeat
14:54:09.253 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 84 9E
14:54:09.256 {"TuyaReceived":{"Data":"55AA0307000803020004000000849E","Cmnd":7,"CmndData":"0302000400000084","DpType2Id3":132,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000084"}}}
14:54:09.259 TYA: fnId=0 is set for dpId=3
14:54:09.261 TYA: RX value 132 from dpId 3 
14:54:09.754 WIF: Checking connection...
14:54:11.371 TYA: Send "55aa00000000ff"
14:54:11.383 DMP: 55 AA 03 00 00 01 01 04
14:54:11.385 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:54:11.388 TYA: Heartbeat
14:54:22.401 TYA: Send "55aa00000000ff"
14:54:22.413 DMP: 55 AA 03 00 00 01 01 04
14:54:22.415 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:54:22.417 TYA: Heartbeat
14:54:24.165 DMP: 55 AA 03 1C 00 00 1E
14:54:24.168 {"TuyaReceived":{"Data":"55AA031C00001E","Cmnd":28}}
14:54:24.185 TYA: Send "55aa001c0008011801030e3618039f"
14:54:24.187 DMP: 55 AA 03 07 00 08 03 02 00 04 00 00 00 84 9E
14:54:24.189 {"TuyaReceived":{"Data":"55AA0307000803020004000000849E","Cmnd":7,"CmndData":"0302000400000084","DpType2Id3":132,"3":{"DpId":3,"DpIdType":2,"DpIdData":"00000084"}}}
14:54:24.191 TYA: fnId=0 is set for dpId=3
14:54:24.193 TYA: RX value 132 from dpId 3 
14:54:29.795 WIF: Checking connection...
14:54:33.369 TYA: Send "55aa00000000ff"
14:54:33.387 DMP: 55 AA 03 00 00 01 01 04
14:54:33.389 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
14:54:33.391 TYA: Heartbeat`

**Comment by g6094199** [comment]
many thx [to](https://forum.iobroker.net/topic/47322/raumthermostat-me81h-31-wifi-tuya-tasmota-flashen) for the many useful insights. 

DLPS seem to be very similar [to](https://templates.blakadder.com/Moes_WHT-HY609.html)  so far...

The unit also has 6 + 2 timers, temp compensation, sensor selection, temp difference start, antifreeze, temp limit, backlight setting, sound setting


up to now i've found:

**BACKLOG  TuyaMCU 11,1;  TuyaMCU 12,9 ; TuyaMCU 13,108 ; TuyaMCU 71, 3; TuyaMCU 72, 2; WebButton1 Power; WebButton2 Lock; WebButton3 Relay;**


```
DpId  
1 : 1/0 Power on/off bool (Tuyasend 1,$var)
2 : set target temp (depending on dpld 4 (mode) set manual temp or auto temp), 3 digits int (10:1) (Tuyasend2 2,$var)
3 : Internal Temp sensor in 3 counts int  (10:1) 
4 :  set mode  enum 	0 = Manual, 1 = Auto/Program, 2 = Away (Tuyasend2 4,$var)
5 : enum
9 : lock screen bool 0 = unlock, 1 = lock
11: ??
15 : Int
19 : Int
101: Int
102: Int
103: bool ??
104: bool ( when enabled, disables 103 when enabled and enables 109)
105: ??
106: enum
107: enum
108: set relay manually bool
109: bool ??
110: enum
```

**Comment by g6094199** [comment]
when i send 

`TuyaSend 103,1` to 103 gets enabled, but when
`TuyaSend 104,1` to enable 104, 104 stays OFF and 103 and 1 switches to OFF (when enabled) and 109 gets enabled every time i try.

any thoughs about this behaviour?!?!
